akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -25,5 +25,12 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val addressUid: Int = ThreadLocalRandom.current.nextInt()
-}
\ No newline at end of file
+  val longAddressUid: Long = {
+    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
```

Comment:
I have created a separate ticket for this: https://github.com/akka/akka/issues/20644